pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1216,23 +1216,21 @@ func buildDefaultTrafficPolicy(env *model.Environment, discoveryType apiv2.Clust
 	}
 }
 
-func altStatName(statPattern string, host string, subset string, dnsDomain string, port *model.Port) string {
-	name := strings.ReplaceAll(statPattern, serviceStatPattern, shortHostName(host, dnsDomain))
+func altStatName(statPattern string, host string, subset string, port *model.Port) string {
+	name := strings.ReplaceAll(statPattern, serviceStatPattern, shortHostName(host))
 	name = strings.ReplaceAll(name, serviceFQDNStatPattern, host)
 	name = strings.ReplaceAll(name, subsetNameStatPattern, subset)
 	name = strings.ReplaceAll(name, servicePortStatPattern, strconv.Itoa(port.Port))
 	name = strings.ReplaceAll(name, servicePortNameStatPattern, port.Name)
 	return name
 }
 
-func shortHostName(host string, dnsDomain string) string {
-	shortHost := strings.TrimSuffix(host, dnsDomain)
-	if parts := strings.Split(dnsDomain, "."); len(parts) > 1 {
-		shortHost += parts[0] // k8s will have namespace.<domain>
+func shortHostName(host string) string {
```

Comment:
The original logic was unnecessarily complicated with domain - it should just take the service and namespace (first two parts).